### PR TITLE
Support decision function for multiclass

### PR DESCRIPTION
**Pull request by PetrovKP**
Add support decision function for multiclass

### CODE REVIEW

**Comment by PetrovKP** [comment]
ead7852e-6a6c-f136-8131-8cdcd4b723a1?

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ead83cac-0e0a-f130-a7de-8cdcd4b723a1?

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/eaf1ae93-a504-f117-8445-8cdcd4b723a1?

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/eaf226b3-453e-f10f-b3f9-8cdcd4b723a1?

File: cpp/daal/include/algorithms/multi_class_classifier/multi_class_classifier_predict_types.h [hunk 1]
```diff
@@ -108,8 +119,74 @@ class DAAL_EXPORT Input : public classifier::prediction::Input
...
+    * \param[in] par     %Parameter of the algorithm
+    * \param[in] method  Computation method
+    * \return Status of checking
+    */
+    services::Status check(const daal::algorithms::Input * input, const daal::algorithms::Parameter * par, int method) const DAAL_C11_OVERRIDE;
+
+protected:
+    using classifier::prediction::Result::check;
+
+    /** \private */
```
Comment (agorshk) [inline cpp/daal/include/algorithms/multi_class_classifier/multi_class_classifier_predict_types.h:172]: protected?

File: cpp/daal/include/algorithms/multi_class_classifier/multi_class_classifier_predict.h [hunk 2]
```diff
@@ -317,9 +339,9 @@ class Batch : public classifier::prediction::Batch
         return new Batch<algorithmFPType, pmethod, tmethod>(*this);
     }
 
-    services::Status allocateResult() DAAL_C11_OVERRIDE
+    virtual services::Status allocateResult() DAAL_C11_OVERRIDE
     {
-        services::Status s = _result->allocate<algorithmFPType>(&input, &parameter, (int)pmethod);
+        services::Status s = static_cast<ResultType *>(_result.get())->allocate<algorithmFPType>(&input, &parameter, (int)pmethod, (int)tmethod);
```
Comment (agorshk) [inline cpp/daal/include/algorithms/multi_class_classifier/multi_class_classifier_predict.h:344]: minor: mix C & C++ data casting style

File: cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_predict_result.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ */
+services::Status Result::check(const daal::algorithms::Input * input, const daal::algorithms::Parameter * parameter, int method) const
+{
+    return checkImpl(input, parameter);
+}
+
+services::Status Result::checkImpl(const daal::algorithms::Input * input, const daal::algorithms::Parameter * parameter) const
+{
+    services::Status s;
+    // services::Status s = classifier::prediction::Result::checkImpl(input, parameter);
```
Comment (agorshk) [inline cpp/daal/src/algorithms/multiclassclassifier/multiclassclassifier_predict_result.cpp:76]: remove?

File: examples/daal/cpp/daal_lnx.lst [hunk 4]
```diff
@@ -18,169 +18,169 @@
...
+        mse_dense_batch                                \
+        zscore_dense_batch                             \
+        minmax_dense_batch                             \
+        ridge_reg_norm_eq_dense_batch                  \
+        ridge_reg_norm_eq_dense_online                 \
+        ridge_reg_norm_eq_dense_distr                  \
+        uniform_dense_batch                            \
+        normal_dense_batch                             \
+        bernoulli_dense_batch                          \
+        enable_thread_pinning                          \
```
Comment (agorshk) [inline examples/daal/cpp/daal_lnx.lst:185]: minor: why the format is changed?

**Review by agorshk — APPROVED** [review]

**Pull request merged**